src/Framework/Framework/Compilation/ViewCompiler/ViewCompilingVisitor.cs

```diff
@@ -264,20 +256,19 @@ protected string CreateControl(ResolvedControl control)
         /// <summary>
         /// Emits binding constructor and returns variable name
         /// </summary>
-        protected ExpressionSyntax ProcessBinding(ResolvedBinding binding)
+        protected Expression ProcessBinding(ResolvedBinding binding)
         {
             return bindingCompiler.EmitCreateBinding(emitter, binding);
         }
 
         /// <summary>
         /// Processes the template.
         /// </summary>
-        protected string CreateTemplate(string builderMethodName)
+        protected string CreateTemplate(string builderMethodName, Delegate compiledDelegate)
         {
-            var templateName = emitter.EmitCreateObject(typeof(DelegateTemplate));
-            emitter.EmitSetProperty(templateName,
-                nameof(DelegateTemplate.BuildContentBody),
-                emitter.EmitIdentifier(builderMethodName));
+            var templateName = emitter.EmitCreateObject(typeof(DelegateTemplate)).Name;
```

Comment:
```suggestion
            var templateName = emitter.EmitValue(new DelegateTemplate { BuildContentBody = compiledDelegate });
```

I think we can even do something like this. No need to compile it and then execute it all over again.

Reply:
Solved, compressed the method too